### PR TITLE
Fix frustum query, clarify transformations.

### PR DESCRIPTION
**Pull request by feuerste**
This PR improves the following:
- The frustum query now consists of an isometry describing the transformation from frustum to global coordinates and a perspective transformation.
- The frustum query got a unit test.
- The isometry now transforms points instead of vectors, which is in accordance with cgmath and nalgebra.
- Clarifies the transformation direction of struct variables of OBB, oriented beam, and frustum.

### CODE REVIEW

File: src/iterator.rs [hunk 1]
```diff
@@ -116,7 +116,7 @@ where
                 batch.position = batch
                     .position
                     .iter()
-                    .map(|p| local_from_global * p)
+                    .map(|p| (local_from_global * &Point3::from_vec(*p)).to_vec())
```
Comment (catevita) [inline src/iterator.rs]: is the initialization directly into a vector not possible?

File: src/math.rs [hunk 2]
```diff
@@ -252,8 +252,8 @@ impl<S: BaseFloat> Isometry3<S> {
 
 #[derive(Debug, Clone)]
 pub struct Obb<S> {
-    pub isometry: Isometry3<S>,
-    pub isometry_inv: Isometry3<S>,
+    pub from_obb: Isometry3<S>,
```
Comment (catevita) [inline src/math.rs]: maybe here a comment like in the struct OrientedBeam would help understand the meaning of the isometry

File: src/math.rs [hunk 3]
```diff
@@ -364,44 +367,46 @@ where
...
-            separating_axes: OrientedBeam::precompute_separating_axes(&isometry.rotation),
-            isometry,
+            corners: OrientedBeam::precompute_corners(&from_beam, &half_extent),
+            separating_axes: OrientedBeam::precompute_separating_axes(&from_beam.rotation),
+            from_beam,
         }
     }
 
-    fn precompute_corners(isometry: &Isometry3<S>, half_extent: &Vector2<S>) -> [Point3<S>; 4] {
+    fn precompute_corners(from_beam: &Isometry3<S>, half_extent: &Vector2<S>) -> [Point3<S>; 4] {
```
Comment (catevita) [inline src/math.rs]: maybe here it is worth mentioning that the "corners"  are defined as the intersection of the beam with z = 0

File: src/math.rs (hunk 2 above)
Comment (feuerste) [reply src/math.rs]: I renamed all transformations, ptal!

File: src/math.rs (hunk 3 above)
Comment (feuerste) [reply src/math.rs]: Done.

**Comment by feuerste** [comment]
Thanks for your review!

**Pull request merged**